src/lib/util/PathMatcher.ts

```diff
@@ -132,6 +134,16 @@ export class PathMatcher {
 	 * @returns {Promise<boolean>}
 	 */
 	public async pathMatchesPatterns(target: string): Promise<boolean> {
-		return this.matcher(normalize(target));
+		return (await this.matcher(normalize(target))) && !this.isDotFileOrIsInDotFolderUnderProjectDir(target);
+	}
+
+	// Note that we do not want to exclude typical files underneath the projectDir just because the projectDir is
```

Comment:
Could this be a JSDoc-style comment?

Reply:
Sure... but i'm not sure why... it is a private function.

Reply:
Done.